opentelemetry-sdk/src/trace/mod.rs

```diff
@@ -136,7 +136,11 @@ mod tests {
             }
         }
 
-        fn on_end(&self, _span: SpanData) {}
+        fn on_end(&self, _span: SpanData) {
+            // TODO: Accessing Context::current() will panic today and hence commented out.
+            // See https://github.com/open-telemetry/opentelemetry-rust/issues/2871
+            // let _c = Context::current();
```

Comment:
We can also add an ignored test using a TestSpanProcessor that accesses Context::current() in on_end(), to validate the expected behavior once it's safe.